### PR TITLE
Use GooGet to package GCE Windows components

### PR DESCRIPTION
**Pull request by adjackura**
- Remove grouped installer code
- Remove Omaha references
- Create individual packages for each component
  - google-compute-engine-windows: Windows agent exe
  - google-compute-engine-windows-common: Windows agent dll
  - google-compute-engine-powershell: PowerShell module for common functions
  - google-compute-engine-metadata-scripts: Metadata scripts launcher
- Create an google-compute-engine-auto-updater package 
  - Uses the disable-agent-updates metadata key to disable auto updates 
- Build 64 bit binaries

### CODE REVIEW

**Comment by illfelder** [comment]
Please fix your pull request so that it is clear what you are changing.

File: agent/GCEWindowsAgent/Service.cs [hunk 1]
```diff
@@ -27,12 +27,12 @@ namespace Google.ComputeEngine.Agent
     public sealed class ComputeEngineService : ServiceBase
     {
         private readonly MetadataService metadataService;
-
+
```
Comment (illfelder) [inline agent/GCEWindowsAgent/Service.cs]: Remove extra spaces.

File: agent/GCEWindowsAgent/Service.cs [hunk 2]
```diff
@@ -27,12 +27,12 @@ namespace Google.ComputeEngine.Agent
...
     {
         private readonly MetadataService metadataService;
-
+        
         public ComputeEngineService()
         {
             metadataService = new MetadataService();
         }
-
+
```
Comment (illfelder) [inline agent/GCEWindowsAgent/Service.cs]: And here.

File: google-compute-engine-windows-common.goospec [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "google-compute-engine-windows-common",
+  "version": "3.5.0@2",
+  "arch": "x86_64",
+  "authors": "Google Inc.",
+  "license": "http://www.apache.org/licenses/LICENSE-2.0",
+  "description": "Google Compute Engine windows agent Common dll",
```
Comment (illfelder) [inline google-compute-engine-windows-common.goospec]: Please fix the capitalization.

File: metadata_scripts/GCEMetadataScripts/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -32,14 +36,13 @@
...
-//      Major Version: Large feature and functionality changes, or incompatible
-//                     API changes.
+//      Major Version: Large feature and functionality changes, or incompatible API changes.
 //      Minor Version: Add functionality without API changes.
 //      Build Number: Bug fixes.
 //      Revision: Minor changes or improvements such as style fixes.
 //
-// You can specify all the values or you can default the Build and Revision
-// Numbers by using the '*' as shown below:
+// You can specify all the values or you can default the Build and Revision Numbers
```
Comment (illfelder) [inline metadata_scripts/GCEMetadataScripts/Properties/AssemblyInfo.cs]: Comments should wrap at 80 characters.

File: google-compute-engine-auto-updater.goospec [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "google-compute-engine-auto-updater",
+  "version": "1.0.0@2",
+  "arch": "noarch",
+  "authors": "Google Inc.",
+  "license": "http://www.apache.org/licenses/LICENSE-2.0",
+  "description": "Google Compute Engine auto updater for core packages",
```
Comment (illfelder) [inline google-compute-engine-auto-updater.goospec]: Google Compute Engine automatic updater for Google provided packages

File: auto_updater_install.ps1 [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+$action.Arguments = '-ExecutionPolicy Bypass -NonInteractive -NoProfile -File "C:\Program Files\Google\Compute Engine\tools\auto_updater.ps1"'
+
+# Run task 5 minutes after boot, then every day indefinitely 
+$boot_trigger = $task.Triggers.Create(8)
+$boot_trigger.Delay = 'PT5M'
+$boot_trigger.Repetition.Interval= 'P1D'
+
+$folder = $ScheduleService.GetFolder('\')
+$folder.RegisterTaskDefinition('Compute Engine Auto Updater',$task,6,'System',$null,5)
+
```
Comment (illfelder) [inline auto_updater_install.ps1]: Remove trailing new line.

File: auto_updater_install.ps1 [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+$action.Path = 'powershell.exe'
+$action.Arguments = '-ExecutionPolicy Bypass -NonInteractive -NoProfile -File "C:\Program Files\Google\Compute Engine\tools\auto_updater.ps1"'
+
+# Run task 5 minutes after boot, then every day indefinitely 
+$boot_trigger = $task.Triggers.Create(8)
+$boot_trigger.Delay = 'PT5M'
+$boot_trigger.Repetition.Interval= 'P1D'
+
+$folder = $ScheduleService.GetFolder('\')
+$folder.RegisterTaskDefinition('Compute Engine Auto Updater',$task,6,'System',$null,5)
```
Comment (illfelder) [inline auto_updater_install.ps1]: "Here and throughout and everywhere in written English: space after commas"

File: google-compute-engine-auto-updater.goospec (hunk 5 above)
Comment (adjackura) [inline google-compute-engine-auto-updater.goospec]: while its all packages right now we will release packages that dont get auto updated, maybe 'core Google packages' ?

File: agent/Common/Common.csproj [hunk 8]
```diff
@@ -1,65 +1,84 @@
...
+  </ItemGroup>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <PropertyGroup>
+    <PostBuildEvent>
             SET DROP_DIR="$(ProjectDir)bin"
             IF NOT EXIST %25DROP_DIR%25 (
             mkdir %25DROP_DIR%25
             )
             copy "$(TargetPath)" "%25DROP_DIR%25\$(TargetFileName)" /y
         </PostBuildEvent>
```
Comment (illfelder) [inline agent/Common/Common.csproj]: This indentation looks wrong. What's going on here?

File: agent/GCEWindowsAgent/ArgumentValidator.cs [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+        /// <summary>
+        /// Throw ArgumentException if argument is null or an empty string.
+        /// </summary>
+        public static void ThrowIfNullOrEmpty(string argument, string paramName)
+        {
+            ThrowIfNull(argument, paramName);
+            if (argument.Equals(string.Empty))
+            {
+                throw new ArgumentException(string.Format("The argument {0} is empty", paramName));
```
Comment (illfelder) [inline agent/GCEWindowsAgent/ArgumentValidator.cs]: Please end the exceptions with periods. Here and below.

File: agent/GCEWindowsAgent/Service.cs [hunk 10]
```diff
@@ -32,7 +32,7 @@ public ComputeEngineService()
...
```
Comment (illfelder) [inline agent/GCEWindowsAgent/Service.cs]: Remove.

**Pull request merged**